packages/aws-lambda-otel-extension/opt/otel-extension/external/helper.js

```diff
@@ -10,6 +11,37 @@ const extensionVersion = (() => {
   return require('../version');
 })();
 
+const resolveJson = (data) => {
+  try {
+    const res = JSON.parse(data);
+    return res;
+  } catch (error) {
+    return null;
+  }
+};
+const isJson = (data) => !!resolveJson(data);
+
+const stripResponseBlobData = (data) => {
+  data.isBodyExcluded = false;
+  const raw = resolveJson(data.responseData);
```

Comment:
Why do we attempt to resolve JSON out of the eventual top level string? This doesn't serve any use case we've discussed (for HTTP API case, it's objects with JSON string at body property that are returned and not JSON strings - do I miss something?)

Reply:
For http app api data it is a JSON string so we need to parse the string before we can inspect the body property. This body property is also stringified. 

If you take a look at the tests that are attached to this PR it describes exactly what the payloads look like for each use case.

Reply:
Hmm.. in my understanding HTTP API response as send from lambda is in this form:

```javascript
{ statusCode: 200, body: '{"message":"ok"}' }
```
And not this:

```javascript
"{\"statusCode\":200,\"body\":\"{\\\"message\\\":\\\"ok\\\"}\"}"
```

Or is returning JSON string from  lambda, is also valid in that case, an it'll be parsed out by AWS (?)

Reply:
I think by the time it gets to our function the payload has been stringified at some point either by an otel library or by us but when I am printing the response payload from this function I am getting the following 👇 

```
"{\"statusCode\":200,\"body\":\"{\\\"message\\\":\\\"ok\\\"}\"}"
```

Reply:
Hmmm.. as I check it doesn't seem to be the case.

1. Otel instrumentation provides us with  Lambda response as-is (so object in case of HTTP API) ->  https://github.com/serverless/runtime/blob/4e329cd5471e878efa30283e7fa44bee874c304e/packages/aws-lambda-otel-extension/opt/otel-extension/internal/index.js#L387
2. We're passing this as-is to external extension -> https://github.com/serverless/runtime/blob/4e329cd5471e878efa30283e7fa44bee874c304e/packages/aws-lambda-otel-extension/opt/otel-extension/internal/index.js#L262
3. ofc for transport whole request payload is serialized -> https://github.com/serverless/runtime/blob/4e329cd5471e878efa30283e7fa44bee874c304e/packages/aws-lambda-otel-extension/opt/otel-extension/internal/index.js#L380 and unserialized in external extension -> https://github.com/serverless/runtime/blob/4e329cd5471e878efa30283e7fa44bee874c304e/packages/aws-lambda-otel-extension/opt/otel-extension/external/initialize-telemetry-listener.js#L25

Then this data is never seralized by us in context of external extension, so what lands in `stripResponseBlobData` is lambda response in exact shape as it was returned by lambda.

I've also doubled confirmed that, by using an extension with extra logs added and deploying this service: https://github.com/medikoo/test-serverless/tree/test-http-api-console-response

Here it's an object as follows that ends as `responseData` in external extension:

```javascript
{ statusCode: 200, body: '{"message":"ok"}' }
```